salt/minion.py

```diff
@@ -1071,13 +1071,18 @@ def _create_minion_object(
         """
         Helper function to return the correct type of object
         """
+        load_grains = False
+        if opts.get("grains", {}):
```

Comment:
```suggestion
        if not opts.get("grains"):
```
Less objects being created.

Reply:
i did not include `not` on purpose. We want to ensure we also load the default grains if custom grains are set in the config on startup of the minion.

Reply:
Hmm, my logic is wrong. It should actually be `if opts.get("grains"):`